torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2822,7 +2824,7 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                     # reduce_dtype matches the param dtype.
                     param.grad.data = param.grad.data.to(self.mixed_precision.reduce_dtype)
 
-                if self.gradient_predivide_factor > 1:
+                if self.gradient_predivide_factor > 1 and self.sharding_strategy != ShardingStrategy.NO_SHARD:
```

Comment:
might be clearer if we just check that a hook is not registered

Reply:
same, we should not change default logic here

Reply:
Noted, adjusted the logic.